### PR TITLE
[python/r] Update lifecycle tags to "Maturing"

### PR DESCRIPTION
**Pull request by johnkerl**
In prep for upcoming spatial-transcriptomics work, which will be tagged experimental.

See also https://github.com/single-cell-data/SOMA/pull/203

### CODE REVIEW

**Comment by eddelbuettel** [comment]
I noticed in the top-level per-file summary that a file `apis/r/tags` was added, is that accidental?

image

**Comment by johnkerl** [comment]
@eddelbuettel https://github.com/single-cell-data/TileDB-SOMA/pull/2807/commits/d9468b73262eba01a613675780f2459b3b82b617

File: apis/python/src/tiledbsoma/_exception.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class SOMAError(Exception):
     """Base error type for SOMA-specific exceptions.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:13]:
```suggestion
    Lifecycle: Maturing.
```

The other files have a '.' after the lifecycle.

File: apis/python/src/tiledbsoma/_exception.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ class DoesNotExistError(SOMAError):
 def is_does_not_exist_error(e: RuntimeError) -> bool:
     """Given a RuntimeError, return true if it indicates the object does not exist
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:31]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class SOMAError(Exception):
 class DoesNotExistError(SOMAError):
     """Raised when attempting to open a non-existent or inaccessible SOMA object.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:22]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 4]
```diff
@@ -56,7 +56,7 @@ def is_does_not_exist_error(e: RuntimeError) -> bool:
 class AlreadyExistsError(SOMAError):
     """Raised when attempting to create an already existing SOMA object.
 
-    Lifecycle: experimental
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:59]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 5]
```diff
@@ -65,7 +65,7 @@ class AlreadyExistsError(SOMAError):
 def is_already_exists_error(e: SOMAError) -> bool:
     """Given a SOMAError, return true if it indicates the object already exists
 
-    Lifecycle: experimental
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:68]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 6]
```diff
@@ -86,7 +86,7 @@ def is_already_exists_error(e: SOMAError) -> bool:
 class NotCreateableError(SOMAError):
     """Raised when attempting to create an already existing SOMA object.
 
-    Lifecycle: experimental
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:89]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 7]
```diff
@@ -95,7 +95,7 @@ class NotCreateableError(SOMAError):
 def is_not_createable_error(e: SOMAError) -> bool:
     """Given a TileDBError, return true if it indicates the object cannot be created
 
-    Lifecycle: experimental
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:98]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 8]
```diff
@@ -132,7 +132,7 @@ def is_duplicate_group_key_error(e: SOMAError) -> bool:
     """Given a TileDBError, return try if it indicates a duplicate member
     add request in a tiledb.Group.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:135]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_exception.py [hunk 9]
```diff
@@ -141,7 +141,7 @@ def is_domain_setting_error(e: SOMAError) -> bool:
     """Given a SOMAError, return whether it attempted to create
     the ArraySchema but the passed in domain was invalid.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_exception.py:144]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_general_utilities.py [hunk 10]
```diff
@@ -16,23 +16,23 @@
 def get_SOMA_version() -> str:
     """Returns semver-compatible version of the supported SOMA API.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_general_utilities.py:19]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_general_utilities.py [hunk 11]
```diff
@@ -16,23 +16,23 @@
...
     """
     return "0.2.0-dev"
 
 
 def get_implementation() -> str:
     """Returns the implementation name, e.g., "python-tiledb".
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_general_utilities.py:27]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_general_utilities.py [hunk 12]
```diff
@@ -16,23 +16,23 @@
...
     """
     return "python-tiledb"
 
 
 def get_implementation_version() -> str:
     """Returns the package implementation version as a semver.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_general_utilities.py:35]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_general_utilities.py [hunk 13]
```diff
@@ -66,15 +66,15 @@ def assert_version_before(major: int, minor: int) -> None:
 def get_storage_engine() -> str:
     """Returns underlying storage engine name, e.g., "tiledb".
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_general_utilities.py:69]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_general_utilities.py [hunk 14]
```diff
@@ -66,15 +66,15 @@ def assert_version_before(major: int, minor: int) -> None:
...
     """
     return "tiledb"
 
 
 def get_libtiledbsoma_core_version() -> str:
     """Returns the version of libtiledb ("core") used by libtiledbsoma.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_general_utilities.py:77]:
```suggestion
    Lifecycle: Maturing.
```

File: apis/python/src/tiledbsoma/_general_utilities.py [hunk 15]
```diff
@@ -94,7 +94,7 @@ def show_package_versions() -> None:
     """Nominal use is for bug reports, so issue filers and issue fixers can be on
     the same page.
 
-    Lifecycle: maturing
+    Lifecycle: Maturing
```
Comment (jp-dark) [inline apis/python/src/tiledbsoma/_general_utilities.py:97]:
```suggestion
    Lifecycle: Maturing.
```

**Review by jp-dark — APPROVED** [review]
LGTM. The formatting on the Python side is a little inconsistent, but I don't know that it matters.

**Pull request merged**